server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -46,29 +63,73 @@ public String typeName() {
             return CONTENT_TYPE;
         }
 
+        @Override
+        public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
+            return new StoredValueFetcher(context.lookup(), NAME);
+        }
+
         @Override
         public Query existsQuery(SearchExecutionContext context) {
-            // This query is not performance sensitive, it only helps assess
-            // quality of the data, so we may use a slow query. It shouldn't
-            // be too slow in practice since the number of unique terms in this
-            // field is bounded by the number of fields in the mappings.
             return new TermRangeQuery(name(), null, null, true, true);
         }
 
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+            throw new IllegalArgumentException(
+                "aggregations on the '"
+                    + typeName()
+                    + "' field are supported for indices created by version "
+                    + IndexVersions.DOC_VALUES_FOR_IGNORED_META_FIELD
+                    + " or higher"
+            );
+        }
+    }
+
+    public static final class IgnoredFieldType extends StringFieldType {
+
+        private IgnoredFieldType() {
+            super(NAME, true, false, true, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
         @Override
         public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
-            return new StoredValueFetcher(context.lookup(), NAME);
+            return new DocValueFetcher(docValueFormat(format, null), context.getForField(this, FielddataOperation.SEARCH));
+        }
+
+        public Query existsQuery(SearchExecutionContext context) {
+            return new FieldExistsQuery(name());
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+            return new SortedSetOrdinalsIndexFieldData.Builder(
+                name(),
+                CoreValuesSourceType.KEYWORD,
+                (dv, n) -> new KeywordDocValuesField(FieldData.toString(dv), n)
+            );
         }
     }
 
-    private IgnoredFieldMapper() {
-        super(FIELD_TYPE);
+    private IgnoredFieldMapper(StringFieldType fieldType) {
+        super(fieldType);
     }
 
     @Override
     public void postParse(DocumentParserContext context) {
-        for (String field : context.getIgnoredFields()) {
-            context.doc().add(new StringField(NAME, field, Field.Store.YES));
+        if (context.indexSettings().getIndexVersionCreated().onOrAfter(IndexVersions.DOC_VALUES_FOR_IGNORED_META_FIELD)) {
```

Comment:
@javanna I added this logic here to deal with the fact that for an index created by a version before 8.14 (the one where we introduce the change) we would like to keep writing stored values so that, given an index, we have _ignored being stored for already existing indices and doc values only for new ones. Does it make sense?

Reply:
Yes this bit makes sense to me, in that it's not field type dependent and covers for the indexing path.